llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7127,6 +7127,20 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
       return new ICmpInst(I.getInversePredicate(), Builder.CreateAnd(A, B),
                           Op1);
 
+    // Transform (~A ^ B) s< ~A --> (A ^ B) s> A,
+    //           (~A ^ B) s> ~A --> (A ^ B) s< A,
+    //           (~A ^ B) s<= ~A --> (A ^ B) s>= A,
+    //           (~A ^ B) s>= ~A --> (A ^ B) s<= A,
+    //           (~A ^ B) u< ~A --> (A ^ B) u< A,
+    //           (~A ^ B) u> ~A --> (A ^ B) u< A,
+    //           (~A ^ B) u<= ~A --> (A ^ B) u>= A,
+    // and       (~A ^ B) u>= ~A --> (A ^ B) <= A
+    if (match(&I, m_c_ICmp(Pred, m_c_Xor(m_Not(m_Value(A)), m_Value(B)),
+                           m_Not(m_Value(C)))) &&
+        !I.isEquality())
```

Comment:
This transform also works with `eq/ne`.
Alive2: https://alive2.llvm.org/ce/z/FEz2U8

Reply:
@I didn't write them because they are handled.
Aliv2: https://godbolt.org/z/qaTh55fsa

Reply:
I will remove the code that handles them.